aws_jupyter_proxy/awsproxy.py

```diff
@@ -378,10 +378,19 @@ def _build_upstream_auth_info(self) -> UpstreamAuthInfo:
 
         :return: the UpstreamAuthInfo instance
         """
-        auth_header_parts = self.upstream_request.headers["Authorization"].split(" ")
 
-        signed_headers = auth_header_parts[2].strip(",").split("=")[1].split(";")
-        _, _, region, service_name, _ = auth_header_parts[1].split("=")[1].split("/")
+        try:
+            auth_header_parts = self.upstream_request.headers["Authorization"].split(
+                " "
+            )
+
+            signed_headers = auth_header_parts[2].strip(",").split("=")[1].split(";")
+            _, _, region, service_name, _ = (
+                auth_header_parts[1].split("=")[1].split("/")
+            )
+        except (KeyError, IndexError):
+            raise HTTPError(400, message=f"Bad Request")
```

Comment:
Is "Bad Request" sufficient? Does security guideline allow you to put some specific message, like, "Malformed Authorization header"?